mllib/src/main/scala/org/apache/spark/ml/classification/OneVsRest.scala

```diff
@@ -196,8 +196,13 @@ final class OneVsRestModel private[ml] (
     }
 
     // output label and label metadata as prediction
+    val predictionMetadata = new MetadataBuilder()
+      .withMetadata(labelMetadata)
+      .putString("name", predictionCol.name)
```

Comment:
This should go through the standard ML attribute constructors to modify labelMetadata:
```
val predictionMetadata = Attribute.fromMetadata(labelMetadata)
  .withName($(predictionCol))
  .toMetadata()
```